pkg/ingester/index/multi.go

```diff
@@ -0,0 +1,123 @@
+package index
+
+import (
+	"time"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/labels"
+
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/grafana/loki/pkg/querier/astmapper"
+	"github.com/grafana/loki/pkg/storage/config"
+)
+
+type periodIndex struct {
+	time.Time
+	idx int // address of the index to use
+}
+
+type Multi struct {
+	periods []periodIndex
+	indices []Interface
+}
+
+func NewMultiInvertedIndex(periods []config.PeriodConfig, indexShards uint32) (*Multi, error) {
+	var (
+		err error
+
+		ii          Interface // always stored in 0th index
+		bitPrefixed Interface // always stored in 1st index
+
+		periodIndices []periodIndex
+	)
+
+	for _, pd := range periods {
+		switch pd.IndexType {
+		case config.TSDBType:
+			if bitPrefixed == nil {
+				bitPrefixed, err = NewBitPrefixWithShards(indexShards)
+				if err != nil {
+					return nil, err
```

Comment:
nit: maybe you could wrap this error with `errors.Wrap` and add more details to the message, like which period erroed and its index type). Motivation:
there's a chain of calls that aren't wrapping errors that will end up here, so it might be hard to identify the root of the problem. The chain is: 
- `GetOrCreateInstance` is invoked at multiple places without wrapping errors and triggers `newInstance`
  - `newInstance` will invoke `index.NewMultiInvertedIndex` without wrapping errors
    - we end up here, with `NewMultiInvertedIndex` invoking `NewBitPrefixWithShards`

Reply:
Good call!